pr-dhl-woocommerce/includes/abstract-pr-dhl-wc-order.php

```diff
@@ -15,243 +15,255 @@
-				echo '<div class="shipment-dhl-icon-container"><span class="shipment-dhl-icon shipment-dhl-icon-service"></span> ' . __( 'Service', 'dhl-for-woocommerce' ) . '</div>';
-				woocommerce_wp_select ( array(
-					'id'          		=> 'pr_dhl_product',
-					'label'       		=> __( 'Service selected:', 'dhl-for-woocommerce' ),
-					'description'		=> '',
-					'value'       		=> $selected_dhl_product,
-					'options'			=> $dhl_product_list,
-					'custom_attributes'	=> array( $is_disabled => $is_disabled )
-				) );
-			echo '</div>';
-			
-			echo '<div class="shipment-dhl-row-container shipment-dhl-row-weight">';
-
-				$weight_units = get_option( 'woocommerce_weight_unit' );
-
-				// Get weight UoM and add in label
-				echo '<div class="shipment-dhl-icon-container"><span class="shipment-dhl-icon shipment-dhl-icon-weight"></span> ' . __( 'Weight', 'dhl-for-woocommerce' ) . '</div>';			
-				woocommerce_wp_text_input( array(
-					'id'          		=> 'pr_dhl_weight',
-					'label'       		=> sprintf( __( 'Estimated shipment weight (%s) based on items ordered: ', 'dhl-for-woocommerce' ), $weight_units),
-					'placeholder' 		=> '',
-					'description'		=> '',
-					'value'       		=> $selected_weight_val,
-					'custom_attributes'	=> array( $is_disabled => $is_disabled ),
-					'class'				=> 'wc_input_decimal' // adds JS to validate input is in price format
-				) );
-			echo '</div>';
+			$main_button = '<button id="dhl-label-button" class="button button-primary button-save-form">' . esc_html__( 'Generate Label', 'dhl-for-woocommerce' ) . '</button>';
 
-			$this->additional_meta_box_fields( $order_id, $is_disabled, $dhl_label_items, $dhl_obj );
+			// Get tracking info if it exists
+			$label_tracking_info = $this->get_dhl_label_tracking( $order_id );
+			// Check whether the label has already been created or not
+			if ( empty( $label_tracking_info ) ) {
+				$is_disabled = '';
 
+				$print_button = '<a href="#" id="dhl-label-print" class="button button-primary" download target="_blank">' . esc_html__( 'Download Label', 'dhl-for-woocommerce' ) . '</a>';
 
-			// A label has been generated already, allow to delete
-			if( empty( $label_tracking_info ) ) {
-				echo $main_button;
 			} else {
-				echo $print_button;
-				echo $delete_label;
+				$is_disabled = 'disabled';
+
+				$print_button = '<a href="' . $this->get_download_label_url( $order_id ) . '" id="dhl-label-print" class="button button-primary" download target="_blank">' . esc_html__( 'Download Label', 'dhl-for-woocommerce' ) . '</a>';
 			}
 
-			wp_enqueue_script( 'wc-shipment-dhl-label-js', PR_DHL_PLUGIN_DIR_URL . '/assets/js/pr-dhl.js', array('jquery'), PR_DHL_VERSION );
-			wp_localize_script( 'wc-shipment-dhl-label-js', 'dhl_label_data', $dhl_label_data );
+			$dhl_label_data = array(
+				'main_button'  => wp_kses_post( $main_button ),
+				'delete_label' => wp_kses_post( $delete_label ),
+				'print_button' => wp_kses_post( $print_button ),
```

Comment:
these are 3 buttons created within the code, not the user, and the label/text for these buttons are already escaped so there is no need to use **wp_kses_post**